### PR TITLE
`[#312]`:Added Gtihub Action and removed CircleCi

### PR DESCRIPTION
**Pull request by abbas-mehdi-naqvi**
### Summary
1. Added Github actions and removed CircleCi from the project.
2. Added `golang.org/x/sys` package to go modules to fix `go:linkname must refer to declared function or variable` error.
3. Fixed lint errors and testcases.
4. Upgraded the golagci-lint version to `v1.52.2`
5. Upgraded the Go version to `v1.18`.
6. Removed deepsource Go test.

### Issue
https://github.com/standup-raven/standup-raven/issues/312

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+
+      - name: Cache node modules
+        uses: actions/cache@v2
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
```
Comment (ayusht2810) [inline .github/workflows/ci.yml:50]: is this correct? because we are using `yarn`

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+          restore-keys: |
+            ${{ runner.os }}-node-
+
+      - name: Install webapp dependencies
+        run: make vendor
```
Comment (ayusht2810) [inline .github/workflows/ci.yml:55]: what does this `make vendor` do?

File: .golangci.yml [hunk 3]
```diff
@@ -47,6 +47,9 @@ issues:
   exclude:
     - 'declaration of "(err|ctx)" shadows declaration at'
   exclude-rules:
+    - path: server/standup/main.go
```
Comment (ayusht2810) [inline .golangci.yml:50]: why we have excluded this file?

File: server/config/main.go [hunk 4]
```diff
@@ -54,16 +54,14 @@ var (
...
-
-	// derived attributes
-	BotUserID       string         `json:"botUserId"`
-	Location        *time.Location `json:"location"`
-	SentryServerDSN string         `json:"sentryServerDSN"`
-	SentryWebappDSN string         `json:"sentryWebappDSN"`
+	Location                *time.Location `json:"location"`        // derived attributes
+	BotUserID               string         `json:"botUserId"`       // derived attributes
+	SentryServerDSN         string         `json:"sentryServerDSN"` // derived attributes
+	SentryWebappDSN         string         `json:"sentryWebappDSN"` // derived attributes
```
Comment (ayusht2810) [inline server/config/main.go:60]: all these fields are present in the sequence. shouldn't we just write a single comment here?

File: go.mod [hunk 5]
```diff
@@ -15,4 +15,48 @@ require (
 	go.uber.org/atomic v1.6.0
 )
 
+require (
+	github.com/blang/semver v3.5.1+incompatible // indirect
+	github.com/davecgh/go-spew v1.1.1 // indirect
+	github.com/dyatlov/go-opengraph v0.0.0-20180429202543-816b6608b3c8 // indirect
```
Comment (ayusht2810) [inline go.mod:21]: why all these packages are added?

File: .github/workflows/ci.yml (hunk 2 above)
Comment (abbas-mehdi-naqvi) [reply .github/workflows/ci.yml:55]: Downloads server dependencies (runs `go  mod download`)

File: .golangci.yml (hunk 3 above)
Comment (abbas-mehdi-naqvi) [reply .golangci.yml:50]: Govet was trying to space optimize a struct which was provided by a package and hence was immutable
Excluded the file after discussion with manoj

File: go.mod (hunk 5 above)
Comment (abbas-mehdi-naqvi) [reply go.mod:21]: All the packages are indirect dependency of the direct dependencies. 
All these got added after the go version was changed from 1.16 to 1.18

File: .github/workflows/ci.yml (hunk 1 above)
Comment (abbas-mehdi-naqvi) [reply .github/workflows/ci.yml:50]: updating the job with yarn.lock

File: .github/workflows/ci.yml (hunk 2 above)
Comment (ayusht2810) [reply .github/workflows/ci.yml:55]: the name of the job here is `Install webapp dependencies`. Is this correct?

File: go.mod (hunk 5 above)
Comment (ayusht2810) [reply go.mod:21]: have we run the command `go mod tidy`?

Comment (abbas-mehdi-naqvi) [reply go.mod:21]: yes

File: .github/workflows/ci.yml (hunk 2 above)
Comment (abbas-mehdi-naqvi) [reply .github/workflows/ci.yml:55]: Fixed the dependency installation github actions

**Review by ayusht2810 — APPROVED** [review]

**Review by avas27JTG — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+name: CI
+
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+
+jobs:
+  ci:
```
Comment (mickmister) [inline .github/workflows/ci.yml:14]: Thoughts on sharing GitHub action config like this plugin is doing here? https://github.com/mattermost/mattermost-plugin-starter-template/blob/master/.github/workflows/ci.yml

**Review by ayusht2810 — APPROVED** [review]

File: .deepsource.toml [hunk 7]
```diff
@@ -1,14 +1,8 @@
+
 version = 1
 
 test_patterns = ["*_test.go"]
 
-[[analyzers]]
-name = "go"
-enabled = true
```
[the remark this answers is not in the document]
Comment (abbas-mehdi-naqvi) [reply .deepsource.toml:7]: We are using Golangci in our project, so had a discussion with Abhishek and he suggested that we don't need to analyzers for Golang

**Pull request merged**